client/signup/main.jsx

```diff
@@ -149,7 +151,14 @@ const Signup = React.createClass( {
 			defer( function() {
 				page( destination );
 			}.bind( this ) );
-		} else {
+		}
+
+		if ( ! user.get() && config.isEnabled( 'oauth' ) ) {
```

Comment:
Store accessors (`get()` and `isEnabled()` in this case) should be (and, in this case, are) inexpensive, but I moved `user.get()` into a var called `userIsLoggedIn` to make the code easier to read here.